torchvision/datasets/imagenet.py

```diff
@@ -118,54 +107,115 @@ def extra_repr(self):
         return "Split: {split}".format(**self.__dict__)
 
 
-def parse_devkit(root):
-    idx_to_wnid, wnid_to_classes = parse_meta(root)
-    val_idcs = parse_val_groundtruth(root)
-    val_wnids = [idx_to_wnid[idx] for idx in val_idcs]
-    return wnid_to_classes, val_wnids
+@contextmanager
+def tmpdir():
+    tmpdir = tempfile.mkdtemp()
+    try:
+        yield tmpdir
+    except:
+        shutil.rmtree(tmpdir)
 
 
-def parse_meta(devkit_root, path='data', filename='meta.mat'):
-    import scipy.io as sio
+def _splitexts(root):
```

Comment:
Same as above.

Reply:
You mean to move this to `utils.py`? I think this is a bit too specific for now, letting it live here is fine for now